src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/data/TicketDataManager.java

```diff
@@ -0,0 +1,294 @@
+
+                statement.close();
+                connection.close();
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public void createTicketMessage(Ticket ticket, Instant creationDate, String message) throws SQLException {
+        executor.execute(() -> {
+            try {
+                if (!getDataSource().isPresent()) throw new SQLException("Creating the Ticket Message in the database failed... the data source is not present.");
+                Connection connection = getDataSource().get().getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATE_TICKET_MESSAGE_QUERY);
+
+                statement.setInt(1, ticket.getId());
+                statement.setTimestamp(2, Timestamp.from(creationDate));
+                statement.setString(3, message);
+
+                //Check if anything was affected, if it wasn't throw an exception.
+                int rowsAffected = statement.executeUpdate();
+                if (rowsAffected == 0) {
+                    throw new SQLException("Creating the Ticket Message in the database failed... no rows were affected.");
+                }
+
+                statement.close();
+                connection.close();
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public CompletableFuture<TreeMap<Long, String>> retrieveTicketMessages(int ticketID) throws SQLException {
+        return cache.lookupTicketMessagesByTicketID(ticketID);
+    }
+
+    public void updateTicketMessage(Ticket ticket, Instant creationDate, String newMessage) throws SQLException {
+        executor.execute(() -> {
+            try {
+                if (!getDataSource().isPresent())
+                    throw new SQLException("Updating the Ticket Message in the database failed... the data source is not present.");
+                Connection connection = getDataSource().get().getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_TICKET_MESSAGE_QUERY);
+
+                statement.setString(1, newMessage);
+                statement.setInt(2, ticket.getId());
+                statement.setTimestamp(3, Timestamp.from(creationDate));
+
+                //Check if anything was affected, if it wasn't throw an exception.
+                int rowsAffected = statement.executeUpdate();
+                if (rowsAffected == 0) {
+                    throw new SQLException("Updating the Ticket Message in the database failed... no rows were affected.");
+                }
+
+                statement.close();
+                connection.close();
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public void deleteTicketMessage(Ticket ticket, Instant creationDate, String message) throws SQLException {
+        executor.execute(() -> {
```

Comment:
Yes! It will work, I'm using [Sponge's Executor Service](https://jd.spongepowered.org/5.1.0/org/spongepowered/api/scheduler/SpongeExecutorService.html) rather than the usual task builder.